node/src/blockchain/blockchain_interface.rs

```diff
@@ -275,40 +307,51 @@ where
                             let transactions: Vec<BlockchainTransaction> = logs
                                 .iter()
                                 .filter_map(|log: &Log| match log.block_number {
-                                    Some(block_number) => {
-                                        let amount: U256 = U256::from(log.data.0.as_slice());
-                                        let gwei_amount = to_gwei(amount);
-                                        gwei_amount.map(|gwei_amount| BlockchainTransaction {
-                                            block_number: u64::try_from(block_number)
-                                                .expect("Internal Error"),
-                                            from: Wallet::from(log.topics[1]),
-                                            gwei_amount,
-                                        })
-                                    }
                                     None => None,
+                                    Some(block_number) => to_gwei(U256::from(
+                                        log.data.0.as_slice(),
+                                    ))
+                                    .map(|gwei_amount| BlockchainTransaction {
+                                        block_number: block_number.as_u64(),
+                                        from: Wallet::from(log.topics[1]),
+                                        gwei_amount,
+                                    }),
                                 })
                                 .collect();
                             debug!(logger, "Retrieved transactions: {:?}", transactions);
                             // Get the largest transaction block number, unless there are no
-                            // transactions, in which case use start_block.
-                            let last_transaction_block =
-                                transactions.iter().fold(start_block, |so_far, elem| {
-                                    if elem.block_number > so_far {
-                                        elem.block_number
-                                    } else {
-                                        so_far
-                                    }
-                                });
+                            // transactions, in which case use end_block, unless get_latest_block()
+                            // was not successful.
+                            let transaction_max_block_number = if transactions.is_empty() {
```

Comment:
This hunk of code seems like something that could live in an outboard method and make the caller clearer.